swatch-core/schemas/event.yaml

```diff
@@ -171,3 +171,24 @@ properties:
     type: string
     existingJavaType: java.util.Optional<String>
     required: false
+  action:
+    description: Custom action to handle the current event.
+    type: string
+    enum:
+      - add
+      - cleanup
+    required: false
+  span_id:
+    description: Span that identifies which process generated the event.
+    type: string
+    required: false
+  start:
+    description: (Only used when action is `cleanup`). Start time range used to query events to clean up.
+    type: string
+    format: date-time
+    required: false
+  end:
+    description: (Only used when action is `cleanup`). End time range used to query events to clean up.
+    type: string
+    format: date-time
+    required: false
```

Comment:
I don't like having this event schema quite so overloaded. It feels like the correct thing to do is to break the cleanup event into its own schema and then update the kafka listener to deserialize to the generic "event" type or the "cleanup" event type depending on `event_type`. See https://docs.spring.io/spring-kafka/docs/current/reference/html/#serdes-type-methods and https://smallrye.io/smallrye-reactive-messaging/4.4.0/concepts/converters/ (Smallrye reference for thinking about how we'll handle when we port this component to quarkus).

Reply:
Refactored the solution to have two objects: the existing Event and the new one CleanUpEvent. See commit: https://github.com/RedHatInsights/rhsm-subscriptions/pull/2456/commits/f459b709251e96e8fe566f933252a0bcfa3e2a9e

Note that we're not delegating the deserialization to Spring Boot Kafka, but to Jackson Object Mapper (I think this s a good news if we want to migrate to Quarkus). So I didn't need to configure Spring Boot to know what class the message needs to serialize to. 

Instead, I configured Jackson with the `@JsonTypeInfo` and `@JsonSubTypes` annotations what cover exactly this use case. These annotations are placed into a new class `BaseEvent` with the common properties of `Event` and `CleanUpEvent`.